control-plane/pkg/reconciler/broker/namespaced_broker.go

```diff
@@ -83,24 +85,19 @@ func (r *NamespacedReconciler) ReconcileKind(ctx context.Context, broker *eventi
 
 	br := r.createReconcilerForBrokerInstance(broker)
 
-	manifest, err := r.getManifest(broker)
+	// Init contract config map in advance otherwise the volume mounts for the dataplane pods fail
+	_, err := br.GetOrCreateDataPlaneConfigMap(ctx)
```

Comment:
Can we move this to the other reconciler before [1]? Like:

```golang
	// Get contract config map.
	contractConfigMap, err := r.GetOrCreateDataPlaneConfigMap(ctx)
	if err != nil {
		return statusConditionManager.FailedToGetConfigMap(err)
	}

        
	if !r.IsReceiverRunning() || !r.IsDispatcherRunning() {
		return statusConditionManager.DataPlaneNotAvailable()
	}
	statusConditionManager.DataPlaneAvailable()
```

on the other reconciler, it works because we do it in [2], however, I'm not convinced it is the right approach even on the existing reconciler.

Proposal:

- Remove [2]
- do the above
- remove the hack
```golang
	// Init contract config map in advance otherwise the volume mounts for the dataplane pods fail
	_, err := br.GetOrCreateDataPlaneConfigMap(ctx)
```


[1]
https://github.com/knative-sandbox/eventing-kafka-broker/blob/7964c30fbd3a1675e082e66472ba89ae4760ac22/control-plane/pkg/reconciler/broker/broker.go#L90-L93

[2]
https://github.com/knative-sandbox/eventing-kafka-broker/blob/7964c30fbd3a1675e082e66472ba89ae4760ac22/control-plane/pkg/reconciler/broker/controller.go#L78-L84

Reply:
Good suggestion. Done!